### PR TITLE
feat: req.file throws filesize error

### LINKED ISSUE #196: fileSize limit on fastify-multipart did not throw error
I don't know if I use it correctly or not but it is just pass through if I passing file with size higher than limit.

```js
/*
 * other code
 */

fastify.post('/upload', async function (req, reply) {
    const data = await req.file({
      limits: {
        fileSize: 1 * 1024 * 1024,  // 1mb limit
      }
    })

    await pump(data.file, fs.createWriteStream(data.filename))

    reply.send()
})
```

It write file to my directory, when I expect it to throw error

* *node version*: 14.15.3
* *fastify version*: 3.9.2
* *fastify multipart version*: 3.3.1
* *os*: WSL

### PR DESCRIPTION
**Pull request by brainrepo**
Feature proposal:

Allow req.file to throw an error when the file is bigger than the limit

closes #196 

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: test/multipart-small-stream.test.js [hunk 1]
```diff
@@ -30,7 +30,7 @@ test('should throw fileSize limitation error on small payload', async function (
     const part = await req.file({ limits: { fileSize: 2 } })
     await sendToWormhole(part.file)
     if (part.file.truncated) {
-      reply.code(500).send()
+      reply.code(413).send()
```
Comment (mcollina) [inline test/multipart-small-stream.test.js:33]: why this change?

File: index.js [hunk 2]
```diff
@@ -504,6 +504,9 @@ function fastifyMultipart (fastify, options, done) {
     let part
     while ((part = await parts()) != null) {
       if (part.file) {
+        if (part.file.truncated) {
+          throw new RequestFileTooLargeError()
```
Comment (mcollina) [inline index.js:508]: are we sure that the only condition for the file being truncated is that it is too large? Maybe it should be a different error.

**Review by mcollina — CHANGES_REQUESTED** [review]
The approach looks good to me, I've left a few notes.

**Comment by climba03003** [comment]
If you specify it in the function `getMultipartFile`, it can not respect the route level options of `throwFileSizeLimit`.

File: test/multipart-small-stream.test.js (hunk 1 above)
Comment (brainrepo) [reply test/multipart-small-stream.test.js:33]: ops, simply my carelessness

**Comment by mcollina** [comment]
A test is needed.

**Review by mcollina — APPROVED** [review]
lgtm

**Comment by brainrepo** [comment]
> can you check why CI is failing?

Yes, I am doing it, the ci fails on windows for `ERR_STREAM_PREMATURE_CLOSE`, but in my local fork the CI is passing.

**Comment by brainrepo** [comment]
> can you check why CI is failing?

I have installed a windows machine on Virtualbox with node 12.21.0 as the CI trying to reproduce the error but everything works fine.

**Review by simoneb — APPROVED** [review]

**Comment by mcollina** [comment]
@StarpTech could you take a look?

**Comment by StarpTech** [comment]
Sorry, I had no time yesterday.

**Pull request merged**